mailbox/api/src/test/java/org/apache/james/mailbox/events/EventDeadLettersContract.java

```diff
@@ -0,0 +1,184 @@
+        USER,
+        MAILBOX_PATH,
+        MAILBOX_ID,
+        Event.EventId.of("6e0dd59d-660e-4d9b-b22f-0354479f47b4"));
+    MailboxListener.MailboxAdded EVENT_2 = new MailboxListener.MailboxAdded(
+        SESSION_ID,
+        USER,
+        MAILBOX_PATH,
+        MAILBOX_ID,
+        Event.EventId.of("6e0dd59d-660e-4d9b-b22f-0354479f47b5"));
+    MailboxListener.MailboxAdded EVENT_3 = new MailboxListener.MailboxAdded(
+        SESSION_ID,
+        USER,
+        MAILBOX_PATH,
+        MAILBOX_ID,
+        Event.EventId.of("6e0dd59d-660e-4d9b-b22f-0354479f47b6"));
+
+    Group GROUP_A = new EventBusContract.GroupA();
+    Group GROUP_B = new EventBusContract.GroupB();
+    Group NULL_GROUP = null;
+    Event NULL_EVENT = null;
+
+    EventDeadLetters eventDeadLetters();
+
+    @Test
+    default void groupsShouldReturnAllStoredGroups() {
+        EventDeadLetters eventDeadLetters = eventDeadLetters();
+        eventDeadLetters.store(GROUP_A, EVENT_1).block();
+        eventDeadLetters.store(GROUP_B, EVENT_1).block();
+
+        assertThat(eventDeadLetters.groups().toStream())
+            .containsExactlyInAnyOrder(GROUP_A, GROUP_B);
+    }
+
+    @Test
+    default void groupsShouldReturnEmptyWhenNoStored() {
+        EventDeadLetters eventDeadLetters = eventDeadLetters();
+
+        assertThat(eventDeadLetters.groups().toStream()).isEmpty();
+    }
+
+    @Test
+    default void storeShouldThrowWhenNullGroup() {
+        EventDeadLetters eventDeadLetters = eventDeadLetters();
+
+        assertThatThrownBy(() -> eventDeadLetters.store(NULL_GROUP, EVENT_1))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    default void storeShouldThrowWhenNullEvent() {
+        EventDeadLetters eventDeadLetters = eventDeadLetters();
+
+        assertThatThrownBy(() -> eventDeadLetters.store(GROUP_A, NULL_EVENT))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    default void storeShouldThrowWhenBothGroupAndEventAreNull() {
+        EventDeadLetters eventDeadLetters = eventDeadLetters();
+
+        assertThatThrownBy(() -> eventDeadLetters.store(NULL_GROUP, NULL_EVENT))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    default void storeShouldStoreGroupWithCorrespondingEvent() {
+        EventDeadLetters eventDeadLetters = eventDeadLetters();
+
+        eventDeadLetters.store(GROUP_A, EVENT_1).block();
+        assertThat(eventDeadLetters.take(GROUP_A).toStream())
+            .containsOnly(EVENT_1);
+    }
+
+    @Test
+    default void storeShouldWorkWhenMultipleDuplicatedStores() {
+        EventDeadLetters eventDeadLetters = eventDeadLetters();
+
+        eventDeadLetters.store(GROUP_A, EVENT_1).block();
+        eventDeadLetters.store(GROUP_A, EVENT_1).block();
+        eventDeadLetters.store(GROUP_A, EVENT_1).block();
+
+        assertThat(eventDeadLetters.take(GROUP_A).toStream())
+            .containsExactly(EVENT_1, EVENT_1, EVENT_1);
```

Comment:
Does this makes sens?

I would expect only `EVENT_1` one time.

Look at HashMultimap.